### DISCUSSION TITLE
without lambda keywords?

### DISCUSSION THREAD

**Discussion by abersheeran** [hsfzxjy/lambdex#3] [Ideas]
Just like:

```python
def_([x, y], [
    return_[x + y]
])
```

**Comment by hsfzxjy** [comment]
Unfortunately this is impossible, since `[x, y]` and `return_[x + y]` will be evaluated eagerly and raise a NameError.  The `lambda`  can prevent the body being evaluated before transpiling.

Besides, the syntax `def_([x, y], ...)` is hard to represent advanced parameter specification like keyword-only arguments or default arguments. `lambda` instead has a consistent parameter specification like ordinary functions. See https://github.com/hsfzxjy/lambdex#parameters
